src/remote.c

```diff
@@ -2162,15 +2162,19 @@ static int remove_branch_config_related_entries(
 		if (git_buf_printf(&buf, "branch.%.*s.merge", (int)branch_len, branch) < 0)
 			break;
 
-		if ((error = git_config_delete_entry(config, git_buf_cstr(&buf))) < 0)
-			break;
+		if ((error = git_config_delete_entry(config, git_buf_cstr(&buf))) < 0) {
+			if (error != GIT_ENOTFOUND)
+				break;
```

Comment:
Could I trouble you to do a `giterr_clear()` here to remove the error message that `git_config_delete_entry` added?  We would like to avoid leaving that error message hanging around when there isn't actually an error.